src/App.js

```diff
@@ -23,21 +49,126 @@ class App extends React.Component {
     }));
   };
 
+  // check for winner using rank
+  // ask MS how to tidy this up
+  checkWinner = () => {
+    if (this.state.currCards.length > 1) {
```

Comment:
instead of putting this logic here, you could put this in the `componentDidUpdate` hook:

```jsx
function componentDidUpdate(prevProps, prevState) {
  if (prevState.currCards !== this.state.currCards && this.state.currCards.length) {
    this.checkWinner();
  }
}
```